File: src/main/python/partitioningIntoMinimumNumberOfDeciBinaryNumbers/PartitioningIntoMinimumNumberOfDeciBinaryNumbers.py
class Solution:
    def minPartitions(self, n: str) -> int:
        return int(max(n))

    def test(self):
        tests = [
            ["32", 3],
            ["82734", 8],
            ["27346209830709182346", 9],
        ]
        for test in tests:
            n, expect_result = test
            result = self.minPartitions(n)
            print(result)
            print(result == expect_result)


if __name__ == '__main__':
    Solution().test()

File: src/main/kotlin/me/bossm0n5t3r/leetcode/partitioningintominimumnumberofdecibinarynumbers/PartitioningIntoMinimumNumberOfDeciBinaryNumbers.kt
package me.bossm0n5t3r.leetcode.partitioningintominimumnumberofdecibinarynumbers

class PartitioningIntoMinimumNumberOfDeciBinaryNumbers {
    class Solution {
        fun minPartitions(n: String): Int = n.max() - '0'
    }
}
